### PR TITLE
MediaBox: convert to stateless

### PR DESCRIPTION
**Pull request by daubaris**
# Checklist:
#928 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by alextrastero** [comment]
Could you merge master into this branch?
`git merge master --no-ff`

File: src/MediaBox.js [hunk 1]
```diff
@@ -1,45 +1,33 @@
...
-    }
-  }
+const MediaBox = ({ id, options, children, className, caption, ...props }) => {
+  const instance = useRef(null);
+  const _id = id || `mediabox${idgen()}`;
 
-  render() {
-    const { children, className, caption, ...props } = this.props;
+  useEffect(() => {
+    instance.current = M.Materialbox.init(
```
Comment (alextrastero) [inline src/MediaBox.js]: `const instance = `

File: test/MediaBox.spec.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ describe('<MediaBox />', () => {
   });
 
   test('renders', () => {
-    expect(shallow(wrapper)).toMatchSnapshot();
+    expect(mount(wrapper)).toMatchSnapshot();
```
Comment (alextrastero) [inline test/MediaBox.spec.js]: sorry to be a pain, does this work with shallow? if so, let's leave shallow

Comment (daubaris) [reply test/MediaBox.spec.js]: It's fine :smile:

**Pull request merged**